hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/BackgroundContainerDataScanner.java

```diff
@@ -110,9 +112,19 @@ public void scanContainer(Container<?> c)
 
   @Override
   public Iterator<Container<?>> getContainerIterator() {
+    recordContainersMetric();
```

Comment:
This is only called at the beginning of a background scan, which on dense volumes may happen once every few weeks. Real time updates as containers are moved, created, deleted, should be plugged directly into the `ContainerSet`.

Reply:
It was very helpful and helped me better understand this part of the code. I’ve improved the code. If you have time, please take a look.